src/replication/ng/policy/manager.go

```diff
@@ -32,3 +120,69 @@ type Manager interface {
 	// Remove the specified policy
 	Remove(int64) error
 }
+
+// DefaultManager provides replication policy CURD capabilities.
+type DefaultManager struct{}
+
+var _ Manager = &DefaultManager{}
+
+// NewDefaultManager is the constructor of DefaultManager.
+func NewDefaultManager() *DefaultManager {
+	return &DefaultManager{}
+}
+
+// Create creates a new policy with the provided data;
+// If creating failed, error will be returned;
+// If creating succeed, ID of the new created policy will be returned.
+func (m *DefaultManager) Create(policy *model.Policy) (int64, error) {
+	ply, err := convertToPersistModel(policy)
+	if err != nil {
+		return 0, err
+	}
+	return dao.AddRepPolicyNg(*ply)
+}
+
+// List returns all the policies
+func (m *DefaultManager) List(query ...*model.PolicyQuery) (int64, []*model.Policy, error) {
+	// TODO: by query args
+	policies, err := dao.GetPoliciesNg()
+	if err != nil {
+		return 0, nil, err
+	}
+
+	var result []*model.Policy
+	for _, policy := range policies {
+		ply, err := convertFromPersistModel(policy)
+		if err != nil {
+			return 0, nil, err
+		}
+
+		result = append(result, ply)
+	}
+	return int64(len(result)), result, nil
```

Comment:
The first returning value of the function should be the total count, but the `result` seems just a subset?

Reply:
ok